api/plugin/file/fileutils.go

```diff
@@ -0,0 +1,65 @@
+package file
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func getRealPath(path string) string {
+	wd, err := os.Getwd()
+	if err != nil {
+		log.Fatal("Error getting working directory:", err)
+	}
+	return filepath.Join(wd[:len(wd)-23], path)
+}
+
+// A valid path should be a shortest path name equivalent to path by purely lexical processingand.
+// Specifically, it should start with "/parigot/app/", also, any use of '.', '..', in the path is
+// not allowed.
+func isValidPath(fpath string) (string, bool) {
+	fileName := filepath.Base(fpath)
+	dir := strings.ReplaceAll(fpath, fileName, "")
+	if !strings.HasPrefix(dir, pathPrefix) || strings.Contains(dir, ".") {
+		return fpath, false
+	}
+	cleanPath := filepath.Clean(fpath)
+
+	return cleanPath, true
+}
+
+func isValidBuf(buf []byte) bool { return len(buf) <= maxBufSize }
+
+// Deletes the specified file and its parent directories if they're empty
+func deleteFileAndParentDirIfNeeded(path string) {
+	realPath := getRealPath(path)
+
+	// Delete the file
+	err := os.Remove(realPath)
+	if err != nil {
+		log.Fatalf("Failed to delete file: %s. Error: %v", path, err)
+	}
+
+	// Walk up the directory tree and remove any empty directories.
+	dir := filepath.Dir(realPath)
+	for {
```

Comment:
1) This behavior with the walking up the directory tree definitely needs a couple of tests to prove it works correctly.

2) So you are going to have "delete" be able to delete directories as well?